app/controllers/incomeSources/add/BusinessReportingMethodController.scala

```diff
@@ -121,62 +120,132 @@ class BusinessReportingMethodController @Inject()(val authenticate: Authenticati
   }
 
   private def handleSubmitRequest(isAgent: Boolean, id: String)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
-    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
-    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
-    val redirectUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAddedObligationsController.showAgent(id) else controllers.incomeSources.add.routes.BusinessAddedObligationsController.show(id)
+    withIncomeSourcesFS {
+      AddBusinessReportingMethodForm.form.bindFromRequest().fold(
+        formWithErrors => handleFormErrors(formWithErrors, id, isAgent),
+        valid => handleFormData(valid, id, isAgent))
+    }
+  }
+
+  private def handleFormErrors(formWithErrors: Form[AddBusinessReportingMethodForm], id: String, isAgent: Boolean)
+                              (implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+
+    val redirectErrorUrl: Call = if (isAgent) routes.IncomeSourceReportingMethodNotSavedController.showBusinessAgent() else
+      routes.IncomeSourceReportingMethodNotSavedController.showBusiness()
     val submitUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.submitAgent(id) else
       controllers.incomeSources.add.routes.BusinessReportingMethodController.submit(id)
 
-    if (incomeSourcesEnabled) {
-      AddBusinessReportingMethodForm.form.bindFromRequest().fold(
-        hasErrors => {
-          val updatedForm = AddBusinessReportingMethodForm.updateErrorMessagesWithValues(hasErrors)
-          getBusinessReportingMethodDetails(id).map {
-            case Some(viewModel) =>
-              BadRequest(view(
-                addBusinessReportingMethodForm = updatedForm,
-                businessReportingViewModel = viewModel,
-                postAction = submitUrl,
-                isAgent = isAgent))
-            case None => Redirect(redirectUrl)
+    val updatedForm = AddBusinessReportingMethodForm.updateErrorMessagesWithValues(formWithErrors)
+    getBusinessReportingMethodDetails(id).map {
+      case Some(viewModel) =>
+        BadRequest(view(
+          addBusinessReportingMethodForm = updatedForm,
+          businessReportingViewModel = viewModel,
+          postAction = submitUrl,
+          isAgent = isAgent))
+      case None => Redirect(redirectErrorUrl)
+    }
+  }
+
+  private def handleFormData(form: AddBusinessReportingMethodForm, id: String, isAgent: Boolean)
+                            (implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    val redirectUrl: Call = if (isAgent) routes.BusinessAddedObligationsController.showAgent(id) else routes.BusinessAddedObligationsController.show(id)
+    val reportingMethodNeedsUpdating = form.taxYear1ReportingMethod != form.newTaxYear1ReportingMethod ||
```

Comment:
can we check and create one TaxYearSpecific at a time ?
As we can update them independently